staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy_test.go

```diff
@@ -136,3 +142,101 @@ func TestPrepareForUpdate(t *testing.T) {
 		}
 	}
 }
+
+var listTypeResourceFixture = &apiextensions.CustomResourceDefinition{
+	ObjectMeta: metav1.ObjectMeta{Name: "foos.test"},
+	Spec: apiextensions.CustomResourceDefinitionSpec{
+		Group: "test",
+		Versions: []apiextensions.CustomResourceDefinitionVersion{
+			{
+				Name:    "v1",
+				Storage: true,
+				Served:  true,
+				Schema: &apiextensions.CustomResourceValidation{
+					OpenAPIV3Schema: &apiextensions.JSONSchemaProps{
+						XPreserveUnknownFields: pointer.BoolPtr(true),
+						Type:                   "object",
+					},
+				},
+			},
+		},
+		Names: apiextensions.CustomResourceDefinitionNames{
+			Plural:   "foos",
+			Singular: "foo",
+			Kind:     "Foo",
+			ListKind: "FooList",
+		},
+		Scope: apiextensions.ClusterScoped,
+	},
+}
+
+const listTypeResourceSchema = `
+type: object
+properties:
+  numArray:
+    type: array
+    XListType: set
+    XMapType: atomic
```

Comment:
this doesn't look right... I think you need to decode into a `k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1#CustomResourceDefinition` object and then convert to a `k8s.io/apiextensions-apiserver/pkg/apis/apiextensions#CustomResourceDefinition` object using `Convert_v1_CustomResourceDefinition_To_apiextensions_CustomResourceDefinition`

Reply:
Fixed.